airflow/models.py

```diff
@@ -343,10 +343,15 @@ def kill_zombies(self, session=None):
                 dag = self.dags[ti.dag_id]
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
+
+                    # now set non db backed vars on ti
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = False
```

Comment:
As the variable `ti` is coming out of a db query and doesn't flow through TIs `__init__` method, cause of that all fields which are not backed by db didn't get initialise and throws error in `ti.get_template_context` later.